### PR TITLE
Keep OptimizationProblem valid

### PR DESCRIPTION
**Pull request by jschueller**
Bounds and constraints have always the dimension of the objective.
They are cleared when the objective is set with a different input dimension.
So, we cannot use an OptimizationProblem to pass a partial problem to (eg GLM), so add accessors to set bounds etc.
Fixes http://trac.openturns.org/ticket/875

### CODE REVIEW

File: lib/src/Base/Optim/OptimizationProblemImplementation.cxx [hunk 1]
```diff
@@ -97,6 +93,12 @@ NumericalMathFunction OptimizationProblemImplementation::getObjective() const
 
 void OptimizationProblemImplementation::setObjective(const NumericalMathFunction & objective)
 {
+  // clear constraint, bounds
+  LOGWARN(OSS() << "Clearing constraints and bounds");
+  equalityConstraint_ = NumericalMathFunction();
+  inequalityConstraint_ = NumericalMathFunction();
+  bounds_ = Interval(0);
+
   clearLevelFunction();
```
Comment (regislebrun) [inline lib/src/Base/Optim/OptimizationProblemImplementation.cxx]: The constraints should be removed only if the new objective function has an incompatible dimension. It allows eg to switch to a finer discretized robust objective function while keeping the constraints as is.

File: lib/src/Uncertainty/Distribution/MeixnerDistribution.cxx [hunk 2]
```diff
@@ -422,19 +422,20 @@ void MeixnerDistribution::update()
...
+  problem.setBounds(getRange());
+  solver_.setStartingPoint(getMean());
   solver_.setProblem(problem);
   solver_.run();
   b_ = std::sqrt(solver_.getResult().getOptimalValue()[0]);
 
   // Define Optimization problem2 : minimization fCD
   problem.setMinimization(true);
   problem.setObjective(fCD);
+  problem.setBounds(getRange());
```
Comment (regislebrun) [inline lib/src/Uncertainty/Distribution/MeixnerDistribution.cxx]: No need to set the bounds another time if the setObjective() method is modified according to my proposal

**Comment by jschueller** [comment]
@regislebrun I implemented your remark

**Comment by sofianehaddad** [comment]
This new behaviour (first objective function then other elements) has to be documented

**Comment by jschueller** [comment]
@sofianehaddad, ok, see OptimizationProblemImplementation_doc.i.in

**Pull request merged**